src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,58 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # TODO: extend the behavior if rotation, tilt axes also need to be checked.
+                # So far checking the above is not required, as we have observed reporting stage position only in x/y
+                # being compromised. The issue with case western system was that the stage metadata was not
+                # updated in timely manner during the overview acquisition in FM mode (during which rotation, tilt are
+                # not changed), resulting in misplaced tiles in the image display. Even when the stage has physically
+                # moved to the target position but took time in updating its actual position through metadata
+                # The issue is resolved by checking if only x and y stage metadata reports the target position properly
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
```

Comment:
Please give a little bit more background information here. I'd write something like:
"The stage is not moving anymore, however in some *rare* cases, the server still reports the old position for a short while (typically < 1s). This can cause all sorts of confusion in the calling code, as it would seem that the move didn't have any effect, and later on, after the position is eventually updated that the stage has moved unexpectedly. So, wait until the reported position is (1) not identical to the starting position and (2) not too far from the target position. "